### PR TITLE
Update DI and cartodb.js dependencies

### PR DESCRIPTION
**Pull request by viddo**
Waypoint for #6508, includes the new `widgetModel.update` API to make widgets to be updated on changes.

@xavijam  ping

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/data/widget-definition-model.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ module.exports = cdb.core.Model.extend({
       return this._attrsForWidgetModel({
         column: o.column,
         operation: o.operation || 'max',
-        syncData: o.sync || true
+        sync_on_data_change: o.sync || true
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/widget-definition-model.js]: So, should we change `o.sync`?

File: lib/assets/javascripts/cartodb3/data/widget-definition-model.js [hunk 2]
```diff
@@ -79,8 +79,8 @@ module.exports = cdb.core.Model.extend({
       return this._attrsForWidgetModel({
         column: o.column,
         aggregation: o.aggregation || 'count',
-        aggregationColumn: o.aggregationColumn || o.column,
-        syncData: o.sync || true
+        aggregation_column: o.aggregation_column || o.column,
+        sync_on_data_change: o.sync || true
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/widget-definition-model.js]: Same here.

File: lib/assets/javascripts/cartodb3/data/widget-definition-model.js [hunk 3]
```diff
@@ -88,14 +88,14 @@ module.exports = cdb.core.Model.extend({
       return this._attrsForWidgetModel({
         column: o.column,
         bins: o.bins || 10,
-        syncData: o.sync || true
+        sync_on_data_change: o.sync || true
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/widget-definition-model.js]: Dixit.

File: lib/assets/javascripts/cartodb3/data/widget-definition-model.js [hunk 4]
```diff
@@ -88,14 +88,14 @@ module.exports = cdb.core.Model.extend({
...
       });
     },
     'time-series': function () {
       var o = this.get('options') || {};
       return this._attrsForWidgetModel({
         column: o.column,
-        syncData: o.sync || true,
+        sync_on_data_change: o.sync || true,
```
Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/widget-definition-model.js]: :dancer:

**Comment by xavijam** [comment]
Just those comments :), :+1:

File: lib/assets/javascripts/cartodb3/data/widget-definition-model.js (hunk 1 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb3/data/widget-definition-model.js]: Yes, and actually I see now that it should be `sync_on_data_change: _.isBoolean(o.sync_on_data_change) ? o.sync_on_data_change : true`, right?

Comment (xavijam) [inline lib/assets/javascripts/cartodb3/data/widget-definition-model.js]: Yep!

**Pull request merged**